src/components/Pagination/Pagination.tsx

```diff
@@ -1,14 +1,156 @@
+    }
+
+    return (
+        <div className={className} onClick={handleClick}>{pageNumber}</div>
+    )
+}
+
+export const Pagination: FC<PatPaginationProps> = (props: any) => {
+    const { count, page, type, color, size, disabled, className, onChange, rowsPerPage, ...rest } = props;
+    const [curPage, setCurPage] = useState<number>(1);
+    const [curRowsPerPage, setCurRowsPerPage] = useState<number>(rowsPerPage || 0);
+    const rowsOptions: Array<number> = [5, 10, 15, 20, 50];
+    let styleClasses = classNames('pagination', {
+        [`page-${type}`]: !disabled && !!type,
+        [`page-${size}`]: !!size,
+        [`page-${color}`]: !disabled && !!color,
+        disabled,
+    });
+    if (className) {
+        styleClasses += ' ' + className;
+    }
+
+    const handleClick = (e: any, cur: number) => {
+        if (!disabled && cur > 0 && cur <= count) {
+            if (!page) {
+                setCurPage(cur);
+            }
+
+            onChange(e, cur, curRowsPerPage);
+        }
+        else {
+            e.preventDefault();
+        }
+    }
+
+    const handleSelectRowsPerPage = (val: number) => {
+        setCurRowsPerPage(val);
+        onChange(null, page ? page : curPage, val);
+    }
+
+    const unControlledComponent = (
+        <div data-testid={"pagination"} className={styleClasses} {...(rest as NativeDivProps)}>
+            <div data-testid={"prev-btn"} id="prev-btn" className="page-icon" onClick={(e) => handleClick(e, curPage - 1)}>
+                <Icon name="angle left" size={size} disabled={disabled || curPage === 1}></Icon>
+            </div>
+
+            {count > 0 ? <PageBtn className={`page-btn ${curPage === 1 ? "selected" : null}`} pageNumber={1} onClick={handleClick}/> : null}
+
+            <div>{curPage > 3 ? "..." : null}</div>
+
+            {curPage > 2 ? <PageBtn className="page-btn" pageNumber={curPage - 1} onClick={handleClick}/> : null}
+            {curPage > 1 && curPage < count ? <PageBtn className="page-btn selected" pageNumber={curPage} onClick={handleClick}/> : null}
+            {curPage < count - 1 ? <PageBtn className="page-btn" pageNumber={curPage + 1} onClick={handleClick}/> : null}
+
+            <div>{curPage < count - 2 ? "..." : null}</div>
+
+            {count > 1 ? <PageBtn className={`page-btn ${curPage === count ? "selected" : null}`} pageNumber={count} onClick={handleClick}/> : null}
+
+            <div data-testid={"next-btn"} className="page-icon" onClick={(e) => handleClick(e, curPage + 1)}>
+                <Icon name="angle right" size={size} disabled={disabled || count === 0 || curPage === count}></Icon>
+            </div>  
+            {
+                rowsPerPage ? 
+                <Dropdown disabled={disabled} className="rows-dropdown" onChange={handleSelectRowsPerPage}>
+                    {rowsOptions.map((r, i) => <Dropdown.Option key={`dropdown-${i}`} active={curRowsPerPage === r} value={r}>{r} rows per page</Dropdown.Option>)}
```

Comment:
Up to personal choice if you want to use 'row' as a more descriptive variable name.
This code makes it quite apparent that r is associated with rows, so it's really not a problem.

Reply:
Using "rows" is not a bad idea but using "rowsPerPage" can be more precise.